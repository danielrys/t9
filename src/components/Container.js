// @flow
import * as React from "react"
import { View, ActivityIndicator } from "react-native"

// styles
import { Colors, Metrics } from "../themes"

import type { ViewType } from "../types"

export type ContainerProps = {
  children?: React.Node,
  style?: ViewType,
  loading: boolean,
  padding?: boolean,
}

const styles = {
  outerWrapper: {
    flex: 1,
  },
  container: {
    backgroundColor: Colors.background,
    flex: 1,
  },
  spinner: {
    alignItems: "center",
    justifyContent: "center",
    flex: 1,
  },
  padding: {
    padding: Metrics.spacings.medium,
  },
}

export default class Container extends React.PureComponent<ContainerProps> {
  static defaultProps = {
    style: {},
    children: null,
    padding: false,
  }

  render() {
    const { children, style, loading, padding } = this.props
    const additionalStyles = padding ? [styles.padding] : []

    return (
      <View style={[styles.outerWrapper, ...additionalStyles]}>
        <View style={[styles.container, style]}>
          {loading ? (
            <View style={styles.spinner}>
              <ActivityIndicator size="large" color={Colors.primary} />
            </View>
          ) : (
            children
          )}
        </View>
      </View>
    )
  }
}
