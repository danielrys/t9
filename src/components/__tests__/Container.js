import React from "react"
import renderer from "react-test-renderer"

import { Text } from "react-native"

// components
import Container from "../Container"

test("renders correctly", () => {
  const tree = renderer
    .create(
      <Container>
        <Text>Hi</Text>
      </Container>,
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
