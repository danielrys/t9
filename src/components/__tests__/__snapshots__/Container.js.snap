// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "rgb(255,255,255)",
          "flex": 1,
        },
        Object {},
      ]
    }
  >
    <Text>
      Hi
    </Text>
  </View>
</View>
`;
